### ISSUE TITLE
MeterFilters and MeterRegistryCustomizers are not ordered

### ISSUE THREAD

**Issue by shakuzen** [spring-projects/spring-boot#14780]
#11843 applies `MeterFilter` beans, but the user cannot control the order in which the `MeterFilter`s are applied, which may matter depending on what all the MeterFilters do.

For instance, I would like to turn on percentile histograms for my application's endpoints, but I want them off for all Actuator endpoints and unmapped requests because this adds ~70 time series for each endpoint per instance (and per status/exception!). My initial thought was to use the property `management.metrics.distribution.percentiles-histogram` like below.

```yaml
management:
  metrics:
    distribution:
      percentiles-histogram:
        http.server.requests: true
```

Then I would only need to make a `MeterFilter` to turn this off if the `uri` tag seemed like an Actuator request (since there isn't a more direct way to do this without #13435) or unmapped request:

```java
@Bean
MeterFilter noActuatorHistogramsFilter() {
	return new MeterFilter() {
		@Override
		public DistributionStatisticConfig configure(Meter.Id id, DistributionStatisticConfig config) {
			if ("http.server.requests".equals(id.getName()) && id.getTag("uri") != null && (id.getTag("uri").contains("actuator") || id.getTag("uri").equals("/**"))) {
				return DistributionStatisticConfig.builder()
						.percentilesHistogram(false)
						.build()
						.merge(config);
			}
			return config;
		}
	};
}
```

Looking at the `MetricsAutoConfiguration` class and seeing that the `PropertiesMeterFilter` has an `@Order(0)`, I thought my custom `MeterFilter` bean would be applied after it, but this was not the case.

https://github.com/spring-projects/spring-boot/blob/a170bfcc76cfb47b1c94481ccd35509eaa619b74/spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/MetricsAutoConfiguration.java#L57-L61

The way I can work around this is to not use the property-based meter filter at all and just put all the logic I want in my own `MeterFilter`, but I think being able to control of the order of `MeterFilter` is a generally good idea regardless of this specific use-case.

**Comment by wilkinsona** [comment]
`MeterRegistryCustomizer` beans have the same problem due to the way both they and `MeterFilter` beans are retrieved by `MeterRegistryPostProcessor`.

This issue's currently targeted at 2.1. I wonder if we should fix it in 2.0.x?

**Comment by philwebb** [comment]
I'll move it back to 2.0.x
